torchvision/prototype/transforms/_augment.py

```diff
@@ -109,16 +113,32 @@ def forward(self, *inpts: Any) -> Any:
             raise TypeError(f"{type(self).__name__}() is only defined for Image's *and* OneHotLabel's.")
         return super().forward(sample)
 
+    def _mixup_onehotlabel(self, inpt: features.OneHotLabel, lam: float) -> features.OneHotLabel:
+        if inpt.ndim < 2:
+            raise ValueError("Need a batch of one hot labels")
+        output = inpt.clone()
+        output = output.roll(1, -2).mul_(1 - lam).add_(output.mul_(lam))
+        return features.OneHotLabel.new_like(inpt, output)
+
 
 class RandomMixup(_BaseMixupCutmix):
     def _get_params(self, sample: Any) -> Dict[str, Any]:
         return dict(lam=float(self._dist.sample(())))
 
     def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
-        if isinstance(inpt, features._Feature):
-            return inpt.mixup(**params)
-        else:
-            return inpt
+        lam = params["lam"]
+        if isinstance(inpt, features.Image):
+            if inpt.ndim < 4:
+                raise ValueError("Need a batch of images")
+            output = inpt.clone()
+            output = output.roll(1, -4).mul_(1 - lam).add_(output.mul_(lam))
+            return features.Image.new_like(inpt, output)
+        if isinstance(inpt, features.OneHotLabel):
+            return self._mixup_onehotlabel(inpt, lam)
+
+        raise TypeError(
+            "RandomMixup transformation does not support bounding boxes, segmentation masks and plain labels"
+        )
```

Comment:
Here BBox and Masks shouldn't be supported, but everything else should be allowed (meta-data).